### PR TITLE
Fix ControlNet outpaint

### PR DESCRIPTION
**Pull request by huchenlei**
## Description
Closes https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/200. Try fix ControlNet outpaint.

Reference: https://github.com/Mikubill/sd-webui-controlnet/discussions/1597
Lama outpaint is only supported in txt2img.

## Screenshots/videos:

Txt2img outpaint outputs:
image
txt2img_lama_outpaint_1
txt2img_lama_outpaint_2

## Checklist:

- [ ] I have read [contributing wiki page](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Contributing)
- [ ] I have performed a self-review of my own code
- [ ] My code follows the [style guidelines](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Contributing#code-style)
- [ ] My code passes [tests](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Tests)

### CODE REVIEW

**Comment by huchenlei** [comment]
I think following code from sd-webui-controlnet get dropped in forge
```python

            # Replace x_t to support inpaint models
            for param in outer.control_params:
                if not isinstance(param.used_hint_cond, torch.Tensor):
                    continue
                if param.used_hint_cond.ndim < 2 or param.used_hint_cond.shape[1] != 4:
                    continue
                if x.shape[1] != 9:
                    continue
                if param.used_hint_inpaint_hijack is None:
                    mask_pixel = param.used_hint_cond[:, 3:4, :, :]
                    image_pixel = param.used_hint_cond[:, 0:3, :, :]
                    mask_pixel = (mask_pixel > 0.5).to(mask_pixel.dtype)
                    masked_latent = outer.call_vae_using_process(process, image_pixel, batch_size, mask=mask_pixel)
                    mask_latent = torch.nn.functional.max_pool2d(mask_pixel, (8, 8))
                    if mask_latent.shape[0] != batch_size:
                        mask_latent = torch.cat([mask_latent.clone() for _ in range(batch_size)], dim=0)
                    param.used_hint_inpaint_hijack = torch.cat([mask_latent, masked_latent], dim=1)
                    param.used_hint_inpaint_hijack.to(x.dtype).to(x.device)
                x = torch.cat([x[:, :4, :, :], param.used_hint_inpaint_hijack], dim=1)
```

**Comment by huchenlei** [comment]
Reference. Without the patch, directly upload the 768x768 lama image and 768x768 mask to inpaint_only.
image

image

**Review by lllyasviel — APPROVED** [review]

**Comment by lllyasviel** [comment]
closed as conflicted
but test related things are good.
if that part is necessary feel free to get another pr
